src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -177,6 +177,31 @@ func logHostsStatus(log logrus.FieldLogger, hosts map[string]inventory_client.Ho
 	log.Infof("Hosts status: %v", hostsStatus)
 }
 
+func (c *controller) updateNodesLabels() bool {
+	ignoreStatuses := []string{models.HostStatusDisabled}
+	ctxReq := utils.GenerateRequestContext()
+	log := utils.RequestIDLogger(ctxReq, c.log)
+
+	assistedNodesMap, err := c.ic.GetHosts(ctxReq, log, ignoreStatuses)
+	if err != nil {
+		log.WithError(err).Error("Failed to get node map from the assisted service")
+		return KeepWaiting
+	}
+
+	for hostname, hostData := range assistedNodesMap {
```

Comment:
At some point the assisted installer may give up on some worker nodes, so you probably want to only loop over the hosts that are not yet in the "Error" state.

Reply:
Understood. So this check should occur when the patch fails and before we retry, correct?

Reply:
Not sure I understand the question, but the gist is that the map should have any hosts in error state removed from it

Reply:
You can also use `ListClusterHostsParams` like in they do in [here](https://github.com/openshift/assisted-installer/blob/89ae514be1699f8b6f5d2e6d6d1ed4ce4e14c2ef/src/inventory_client/inventory_client.go#L265) for the role, but for the "status" of the host instead, to get the service to do the filtering for you